snippets/PowerShell.json

```diff
@@ -161,6 +161,23 @@
       "}"
     ]
   },
+  "Foreach with Progress": {
+		"prefix": "foreach-progress",
+    "description": "Insert a foreach loop with Write-Progress initialized",
+		"body": [
+		  "\\$total = $${1:array}.count",
```

Comment:
Does the total need to be maintained outside the array? Why not just use array.count inside the loop?
```suggestion
		  
```

Reply:
```suggestion
\\$progPercent = \"{0:n2}\" -f ([math]::round(\\$i/\\$array.count,4) * 100)",

Write-Progress -Activity \"${3:activityName}\" -Status \"\\$i of \\$array.count - \\$progPercent% Complete:\" -PercentComplete \\$progPercent",
```